docs/source/overview.rst

```diff
@@ -0,0 +1,136 @@
+Overview
+========
+
+Purpose
+-------
+
+It's easiest to describe The MozDef Event Framework as a set of micro-services you can use to integrate event sources with the `Mozilla Enterprise Defense platform <https://mozdef.readthedocs.io/en/latest/>`_ (MozDef).
+
+Provides
+--------
+
+Many resources are not wholly contained within the datacenter, there are SAAS and IAAS providers which may provide event logging that is external to the network where your SIEM resides. 
+Exposing your security infrastructure doesn't ensure a secure pipeline. In order to solve for this we decided on pulling that information as our method of choice, as opposed to pushing.
+Utilizing cloud based microservices can reduce exposure, ensure efficiency and scalability, and leads to less maintenance.
+
+We created a framework using AWS Cloudformation and the Serverless framework to build a pipeline that will allow placement of lambda scripts specific for an event source and provide continuous deployment and integration.
+This takes the guesswork out of having to write serverless and cloudformation code every time  you want to deploy a pipeline for a new event source and creates a standard method to be used to obtain those events. 
+
+Webhook sources utilize an AWS API gateway that they can post events to. As we scale this out the capabilities will increase.
+The hope is to have a scalable framework that can be deployed for both REST and webhook based sources.
+
+Flow
+----
+
+There are There cloudformation templates to choose from in the templates directory:
+
+* Github repo used as source: templates/codepipeline-cf-template-github-source.yml
+* CodeCommit repo used as the config source: templates/codepipeline-cf-template-codecommit-source.yml
+* Github and CodeCommit used as a multiple source that can be merged: templates/multi_source_template/codepipeline-cf-template-with-merge.yml
+
+We've added toggles to view the basic workflow diagrams below, feel free to choose what works best for you.
+
+Workflow Diagrams
+*****************
+
+.. toggle-header::
+    :header: **1. Using a Single Source Repo:**
+
+      Workflow using a Single Source Repo Cloudformation Template
+
+      .. image:: images/MozDef-Event-Framework_process_flow_single-source.png
+         :width: 600
+         :alt: High Level Overview of the Work/Process Flow Steps From Start to Finish
+
+
+.. toggle-header::
+    :header: **2. Using Multiple Source Repos:**
+
+      Workflow using the Mutliple Source Repos w/ Merge Cloudformation Template
```

Comment:
Thanks, will review and resolve!

Reply:
I'm still on the fence on whether this is the best method for this, it feels clunky to put config data in an s3 bucket that is going into a codecommit repo, when it should only just live in the codecommit repo (this prevents possibly creating an s3 bucket with lax perms if secrets are ever placed in those configs for testing purposes or some other reason)
Lets revisit? But I'll go ahead and modify the diagram for now.

Reply:
Diagram modified

Reply:
AFAIK, there is no other way to create a codecommit repo with some files, other than putting the files in a zip archive in an S3 bucket first.

We could just have the CF template create repo without any code/files, or we could ditch the idea of creating the codecommit repo using CloudFormation completely (so it becomes a manual task as in the single source scenario).

If we choose the first option, the very first app deployment after the CI stack is deployed will fail (as there will be no config files). If we create it manually beforehand I think it will be OK (and is probably the best option)